packages/x-date-pickers/src/models/adapters.ts

```diff
@@ -0,0 +1,586 @@
+   * @param {TDate} value The original date.
+   * @returns {TDate} The end of the day.
+   */
+  endOfDay(value: TDate): TDate;
+  /**
+   * Add the specified number of years to the given date.
+   * @template TDate
+   * @param {TDate} value The date to be changed.
+   * @param {number} amount The amount of years to be added.
+   * @returns {TDate} The new date with the years added.
+   */
+  addYears(value: TDate, amount: number): TDate;
+  /**
+   * Add the specified number of months to the given date.
+   * @template TDate
+   * @param {TDate} value The date to be changed.
+   * @param {number} amount The amount of months to be added.
+   * @returns {TDate} The new date with the months added.
+   */
+  addMonths(value: TDate, amount: number): TDate;
+  /**
+   * Add the specified number of weeks to the given date.
+   * @template TDate
+   * @param {TDate} value The date to be changed.
+   * @param {number} amount The amount of weeks to be added.
+   * @returns {TDate} The new date with the weeks added.
+   */
+  addWeeks(value: TDate, amount: number): TDate;
+  /**
+   * Add the specified number of days to the given date.
+   * @template TDate
+   * @param {TDate} value The date to be changed.
+   * @param {number} amount The amount of days to be added.
+   * @returns {TDate} The new date with the days added.
+   */
+  addDays(value: TDate, amount: number): TDate;
+  /**
+   * Add the specified number of hours to the given date.
+   * @template TDate
+   * @param {TDate} value The date to be changed.
+   * @param {number} amount The amount of hours to be added.
+   * @returns {TDate} The new date with the hours added.
+   */
+  addHours(value: TDate, amount: number): TDate;
+  /**
+   * Add the specified number of minutes to the given date.
+   * @template TDate
+   * @param {TDate} value The date to be changed.
+   * @param {number} amount The amount of minutes to be added.
+   * @returns {TDate} The new date with the minutes added.
+   */
+  addMinutes(value: TDate, amount: number): TDate;
+  /**
+   * Add the specified number of seconds to the given date.
+   * @template TDate
+   * @param {TDate} value The date to be changed.
+   * @param {number} amount The amount of seconds to be added.
+   * @returns {TDate} The new date with the seconds added.
+   */
+  addSeconds(value: TDate, amount: number): TDate;
+  /**
+   * Get the year of the given date.
+   * @template TDate
+   * @param {TDate} value The given date.
+   * @returns {number} The year of the given date.
+   */
+  getYear(value: TDate): number;
+  /**
+   * Get the month of the given date.
+   * @template TDate
+   * @param {TDate} value The given date.
+   * @returns {number} The month of the given date.
+   */
+  getMonth(value: TDate): number;
+  /**
+   * Get the date (e.g: the day in the month) of the given date.
+   * @template TDate
+   * @param {TDate} value The given date.
+   * @returns {number} The date of the given date.
+   */
+  getDate(value: TDate): number;
```

Comment:
Nitpick: This method name always puzzled me... Why `date`, when it's working with `day`... 🤷 🙈 
We can't change the method name now, but maybe we could at least provide a correct JSDoc? 🤔 

P.S. [Dayjs](https://day.js.org/docs/en/get-set/date#docsNav) also has this confusing difference, but at least their doc specifies the correct output.

Reply:
`date` comes from `new Date().getDate()` (which is not a great naming imho).

We can rename to a more explicit naming in v7 (`getMonthDay` for example).

Reply:
IMHO, I'd be in favor of 🙈 
- `getDay`;
- `getDayOfWeek`, `getDayOfTheWeek`, or `getWeekDay`;